pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -430,6 +438,177 @@ func consumerRefMatches(consumer *corev1.ObjectReference, machine *machinev1.Mac
+	if len(encodedUserDataBytes) == 0 {
+		err := fmt.Errorf("could not fetch data from the key: 'userData' in the secret: %s", config.UserData.Name)
+		log.Printf(err.Error())
+		return err
+	}
+
+	var userData []byte
+	userData, err = base64.StdEncoding.DecodeString(string(encodedUserDataBytes))
+	if err != nil {
+		log.Printf("Error base64 decoding userData from %s: %s", config.UserData.Name, err.Error())
+		return fmt.Errorf("error base64 decoding userData from %s : %w", config.UserData.Name, err)
+	}
+
+	// Define an ignition data struct to unmarshal incoming json
+	type IgnitionData struct {
+		Ignition struct {
+			Config struct {
+				Append []struct {
+					Source string
+				}
+			}
+			Security struct {
+				TLS struct {
+					CertificateAuthorities []struct {
+						Source string
+					}
+				}
+			}
+		}
+	}
+	var ignData IgnitionData
+
+	if err := json.Unmarshal(userData, &ignData); err != nil {
+		log.Printf("Cannot unmarshal ignition config from userData : %s, %w", config.UserData.Name, err)
+		return fmt.Errorf("cannot unmarshal ignition config from userData : %s, %w", config.UserData.Name, err)
+	}
+
+	var ignitionURL, caCertRaw string
+	// Fetch IgnitionURL and the CA cert
+	if len(ignData.Ignition.Config.Append) > 0 && len(ignData.Ignition.Config.Append[0].Source) > 0 {
+		ignitionURL = ignData.Ignition.Config.Append[0].Source
+	} else {
+		// We didn't get any URL
+		err := fmt.Errorf("error extracting full Ignition URL from content of %s", config.UserData.Name)
+		log.Printf(err.Error())
+		return err
+	}
+
+	if len(ignData.Ignition.Security.TLS.CertificateAuthorities) > 0 && len(ignData.Ignition.Security.TLS.CertificateAuthorities[0].Source) > 0 {
+		caCertRaw = ignData.Ignition.Security.TLS.CertificateAuthorities[0].Source
+	} else {
+		// We didn't get the URL's Cert
+		err := fmt.Errorf("CA cert for full Ignition URL was not specified in %s", config.UserData.Name)
+		log.Printf(err.Error())
+		return err
+	}
+	caCertB64 := strings.TrimPrefix(caCertRaw, "data:text/plain;charset=utf-8;base64,")
+
+	caCert, err := base64.StdEncoding.DecodeString(caCertB64)
+	if err != nil {
+		log.Printf("Could not base64 decode caCert in %s: %s", config.UserData.Name, err.Error())
+		return fmt.Errorf("could not base64 decode caCert in %s: %w", config.UserData.Name, err)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+	client := retryablehttp.NewClient()
+	client.HTTPClient.Transport = &http.Transport{
+		TLSClientConfig: &tls.Config{
+			RootCAs: caCertPool,
+		},
+	}
+
+	// Get the full ignition from the URL
+	resp, err := client.Get(ignitionURL)
+	if err != nil {
+		log.Printf("Error fetching full ignition from %s : %s", ignitionURL, err.Error())
+		return fmt.Errorf("error fetching full ignition from %s: %w", ignitionURL, err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		log.Printf("Received HTTP status code: %d while fetching full ignition from %s : %s", resp.StatusCode, ignitionURL, err.Error())
```

Comment:
Looks like you're still calling `err.Error()` on a `nil` error. Since the error handling is above, I think you should just remove that from this log statement. Also need to remove it from the line below.

Reply:
+1. That was an oversight. There are so many error returns in this function, I missed this one. Fixed in the latest commit.